client/post-editor/post-editor.jsx

```diff
@@ -408,6 +409,35 @@ export const PostEditor = React.createClass( {
 		return this.state.isSaveBlocked || ! this.state.isEditorInitialized;
 	},
 
+
+	isGoogleDocType: function( type ) {
+		// types identified for gdocs:
+		// "application/x-vnd.google-docs-image-clip+wrapped"
+		// "application/x-vnd.google-docs-document-slice-clip+wrapped"
+		return /google-docs/.test( type );
+	},
+
+	hasTextBeenCopiedFromGoogleDocs: function( e ) {
+		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// types should be an array containing the available formats that were set on copy.
+		// Some browsers, as Firefox, return a DOMStringList instead, see
+		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		const orgTypes = e.clipboardData.types;
+		let types = [];
+		if ( orgTypes.contains ) {
+			types = Array.from( orgTypes );
```

Comment:
>It seems that we can't use `Array.from` yet:

We can. The static object methods from "base" types are fine to use and are provided from `babel-plugin-transform-runtime`:

```
$ babel --plugins transform-runtime <<< "Array.from( [ 1, 2, 3 ] );"
import _Array$from from "babel-runtime/core-js/array/from";
_Array$from([1, 2, 3]);
```

It's the prototype methods we can't use, e.g. `Array#includes`. We need `babel-polyfill` for those.